### PR TITLE
add vit pretrain code

### PR DESCRIPTION
**Pull request by GuoxiaWang**
* add configs/vision_transformer/vit-base-p16-pt_in1k-224_4n32c.yaml
* add `return_type` param for `ToRGB` in passl/datasets/preprocess/transforms.py
* add  `RandCropImage` and `ResizeImage` in passl/datasets/preprocess/transforms.py
* modify `ViTWrapper`, remove head warp,  add google vit style label smoothing for sigmoid cross entropy
* add `GoogleVisionTransformer` model
* add `ViTLRScheduler` lr scheduler, include `cosine` and `linear`

Validate Epoch [300] acc1 (74.852), acc5 (90.396)

### CODE REVIEW

File: passl/modeling/architectures/ViTWrapper.py [hunk 1]
```diff
@@ -36,41 +35,82 @@ def __init__(self,
...
             return self.test_iter(*inputs, **kwargs)
         elif mode == 'extract':
-            return self.backbone(*inputs)
+            return self.backbone.forward_features(x)
         else:
             raise Exception("No such mode: {}".format(mode))
 
-   
-    
     def validation_step(self, val_batch, idx):
```
Comment (firestonelib) [inline passl/modeling/architectures/ViTWrapper.py:91]: validation_step function is used for CLIP, you can detele it.

**Review by firestonelib — APPROVED** [review]
LGTM
